prql/src/ast.rs

```diff
@@ -136,9 +127,50 @@ pub struct NamedArg {
 }
 
 #[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
-pub struct Assign {
-    pub lvalue: Ident,
-    pub rvalue: Box<Item>,
+pub struct NamedExpr {
+    pub alias: Option<Ident>,
+    pub expr: Box<Item>,
+}
+
+impl NamedExpr {
+    pub fn unnamed(expr: Item) -> Self {
+        NamedExpr {
+            alias: None,
+            expr: Box::new(expr),
+        }
+    }
+
+    /// Almost all function parameters and list items are wrapped into NamedExpr, which
+    /// can be passed either to this function or to into_list_with_names.
+    pub fn discard_name(self) -> Result<Item> {
```

Comment:
What were your thoughts about having everything be a `NamedExpr`, rather than having an `Item::NamedExpr` type? Both could work; I realize lots of things can have aliases, though is it a simpler construction to have _everything_ be potentially aliased?

Reply:
Well, we do have `Item::NamedExpr` (that contains `struct NamedExpr`).

The functions on the `NamedExpr` are here because of typing: they are only applicable to `NamedExpr`, so it would not make sense to pollute `Item::` namespace (and `match self {  Item::NamedExpr() }` in each function).

Reply:
Just now I've changed it back: it is much nicer not to have everything wrapped into `NamedExpr`. It does provide some nice typing, so you have to discard possible name, but AST should not have such useless wrappers.